src/interfaces.js

```diff
@@ -73,5 +73,46 @@ interface RouteConfig {
   */
   caseSensitive?: boolean;
 
+  /**
+  * Add to specify an activation strategy if the strategy is always the same and you don't want that
+  * to be in your view-model code. Available values are 'replace' and 'invoke-lifecycle'.
+  */
+  activationStrategy?: string;
+  
   [x: string]: any;
 }
+
+interface RoutableComponent {
+  /**
+  * Implement this hook if you want to control whether or not your view-model can be navigated to.
+  * Return a boolean value, a promise for a boolean value, or a navigation command.
+  */
+  canActivate?: (params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction) => boolean|Promise<boolean>|PromiseLike<boolean>|NavigationCommand;
+
+  /**
+  * Implement this hook if you want to perform custom logic just before your view-model is displayed.
+  * You can optionally return a promise to tell the router to wait to bind and attach the view until
+  * after you finish your work.
+  */
+  activate?: (params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction) => Promise<any>|PromiseLike<any>|IObservable|void;
+
+  /**
+  * Implement this hook if you want to control whether or not the router can navigate away from your
+  * view-model when moving to a new route. Return a boolean value, a promise for a boolean value,
+  * or a navigation command.
+  */
+  canDeactivate?: () => boolean|Promise<boolean>|PromiseLike<boolean>|NavigationCommand;
+
+  /**
+  * Implement this hook if you want to perform custom logic when your view-model is being
+  * navigated away from. You can optionally return a promise to tell the router to wait until
+  * after you finish your work.
+  */
+  deactivate?: () => Promise<any>|PromiseLike<any>|IObservable|void;
+
+  /**
+  * Implement this hook if you'd like to give hints to the router about the activation when reusing
+  * a view model for different routes. Available values are 'replace' and 'invoke-lifecycle'.
+  */
+  determineActivationStrategy?: (params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction) => string;
```

Comment:
Would the build (with Babel) still produce valid output If You'd replace the result type `string` with `'replace' | 'invoke-lifecycle'` to narrow down the return type to the minimal correct options?